app/src/test/java/org/simple/clinic/summary/updatephone/UpdatePhoneNumberDialogControllerTest.kt

```diff
@@ -84,51 +78,69 @@ class UpdatePhoneNumberDialogControllerTest {
   }
 
   @Test
-  @Parameters(method = "validation errors")
-  fun `when save is clicked, the number should not be saved if it's invalid`(
-      validationError: Result
-  ) {
+  fun `when save is clicked, the number should not be saved if it's blank and an error should be shown`() {
+    val newNumber = ""
+    val existingPhoneNumber = TestData.patientPhoneNumber(
+        uuid = UUID.fromString("0e4bf753-009b-4cd6-ae30-aa9935bf2ea6"),
+        patientUuid = patientUuid,
+        number = "old-number"
+    )
+
+    whenever(validator.validate(newNumber, type = LANDLINE_OR_MOBILE)).thenReturn(Blank)
```

Comment:
One more thing which we are supposed to do as part of the Mobius migration is replace these mocks with actual objects as much as possible. Can we do this for the validator?

Reply:
Ah yes, forgot about that.